mavros_extras/src/plugins/landing_target.cpp

```diff
@@ -0,0 +1,198 @@
+/**
+ * @brief Landing target plugin
+ * @file landing_target.cpp
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2015 Nuno Marques.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros/setpoint_mixin.h>
+#include <pluginlib/class_list_macros.h>
+#include <boost/math/constants/constants.hpp>
+
+#include <tf/transform_broadcaster.h>
+#include <geometry_msgs/PoseStamped.h>
+#include <geometry_msgs/PoseWithCovarianceStamped.h>
+
+namespace mavplugin {
+/**
+ * @brief Landing Target plugin
+ *
+ * This plugin is intended to publish the location of a landing area captured from a downward facing camera
+ * to the FCU and/or receive IRLock infrared tracking data.
+ */
+class LandingTargetPlugin : public MavRosPlugin,
+	private TFListenerMixin<LandingTargetPlugin> {
+public:
+	LandingTargetPlugin() :
+		sp_nh("~landing_target"),
+		uas(nullptr),
+		tf_rate(10.0),
+		send_tf(true),
+		listen_tf(false)
+	{ };
+
+	void initialize(UAS &uas_)
+	{
+		uas = &uas_;
+
+		sp_nh.param("send_tf", send_tf, true);
+		sp_nh.param("listen_tf", listen_tf, false);
+		sp_nh.param<std::string>("frame_id", frame_id, "landing_target");
+		sp_nh.param<std::string>("child_frame_id", child_frame_id, "camera_center");
+		sp_nh.param("tf_rate_limit", tf_rate, 50.0);
+
+		land_target_pub = sp_nh.advertise<geometry_msgs::PoseStamped>("landing_target", 10);
+
+		if (listen_tf) {
+			ROS_INFO_STREAM_NAMED("landing_target", "Listen to landing_target transform " << frame_id
+											<< " -> " << child_frame_id);
+			tf_start("LandingTargetTF", &LandingTargetPlugin::send_landing_target);
+		}
+
+		land_target_sub = sp_nh.subscribe("target", 10, &LandingTargetPlugin::land_target_cb, this);
+		local_position_sub = sp_nh.subscribe("/mavros/position/local", 10, &LandingTargetPlugin::local_pos_cb, this);
```

Comment:
I don't like that, at lest it broke node renaming.
I don't decide best way yet, but at least use `~position/local` now.